src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/ArtifactDigestField.tsx

```diff
@@ -0,0 +1,187 @@
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { useCallback } from 'react'
+import { Menu } from '@blueprintjs/core'
+import { FormInput, getMultiTypeFromValue, Layout, MultiTypeInputType, Text } from '@harness/uicore'
+import { get, memoize } from 'lodash-es'
+import type { GetDataError } from 'restful-react'
+
+import { ConfigureOptions } from '@common/components/ConfigureOptions/ConfigureOptions'
+import { EXPRESSION_STRING } from '@pipeline/utils/constants'
+import { useMutateAsGet } from '@common/hooks'
+
+import { Failure, useGetLastSuccessfulBuildForDocker } from 'services/cd-ng'
+import { useStrings } from 'framework/strings'
+
+import css from '../../ArtifactConnector.module.scss'
+
+const onTagInputFocus = (e: React.FocusEvent<HTMLInputElement>, fetchDigest: () => void): void => {
+  if (e?.target?.type !== 'text' || (e?.target?.type === 'text' && e?.target?.placeholder === EXPRESSION_STRING)) {
+    return
+  }
+  fetchDigest()
+}
+
+export function NoDigestResults({
+  digestError,
+
+  defaultErrorText
+}: {
+  digestError: GetDataError<Failure | Error> | null
+  defaultErrorText?: string
+}): JSX.Element {
+  const { getString } = useStrings()
+
+  const getErrorText = useCallback(() => {
+    return defaultErrorText || getString('pipeline.artifactsSelection.errors.nodigest')
+  }, [getString])
+
+  return (
+    <Text className={css.padSmall} lineClamp={1}>
+      {get(digestError, 'data.message', null) || getErrorText()}
+    </Text>
+  )
+}
+
+function ArtifactDigestField({
+  formik,
+  buildDetailsLoading,
+  expressions,
+  allowableTypes,
+  isReadonly,
+  accountIdentifier,
+  orgIdentifier,
+  projectIdentifier,
+  connecterRefValue
+}: any): React.ReactElement {
+  const { getString } = useStrings()
+
+  const loadingPlaceholderText = getString('pipeline.artifactsSelection.loadingDigest')
+  const digestDefaultValue = [{ label: loadingPlaceholderText, value: loadingPlaceholderText }]
+  const [digestItems, setDigestItems] = React.useState<any>(digestDefaultValue)
+  const { data, loading, refetch, error } = useMutateAsGet(useGetLastSuccessfulBuildForDocker, {
+    queryParams: {
+      imagePath: formik?.values?.imagePath,
+      connectorRef: connecterRefValue,
+      accountIdentifier,
+      orgIdentifier,
+      projectIdentifier
+    },
+    requestOptions: {
+      headers: {
+        'content-type': 'application/json'
+      }
+    },
+    lazy: true,
+    body: {
+      tag: formik?.values?.tag?.value
+    }
+  })
+
+  React.useEffect(() => {
+    if (data && !loading) {
+      /* istanbul ignore next */
+      const options = [
+        {
+          label: data && data?.data && data?.data?.metadata && data?.data?.metadata?.SHA,
```

Comment:
This may add `undefined` for label and value as options. Can we handle that case?
```suggestion
          const options = []
      if (data && data?.data && data?.data?.metadata && data?.data?.metadata?.SHA) {
        options.push({ label: data.data.metadata.SHA, value: data.data.metadata.SHA })
      }

      if (data && data?.data && data?.data?.metadata && data?.data?.metadata?.SHAV2) {
        options.push({ label: data.data.metadata.SHAV2, value: data.data.metadata.SHAV2 })
      }

      setDigestItems(options)
```